app/models/theme.rb

```diff
@@ -828,23 +836,44 @@ def scss_variables
     contents
   end
 
-  def convert_settings
-    settings.each do |setting|
-      setting_row = ThemeSetting.where(theme_id: self.id, name: setting.name.to_s).first
+  def migrate_settings
```

Comment:
Small note but I think this method needs to be wrapped in a transaction by default too.